datanode/sqlstore/blocks.go

```diff
@@ -112,7 +112,17 @@ func (bs *Blocks) setLastBlock(b entities.Block) {
 func (bs *Blocks) GetOldestHistoryBlock(ctx context.Context) (entities.Block, error) {
 	defer metrics.StartSQLQuery("Blocks", "GetOldestHistoryBlock")()
 
-	return GetOldestHistoryBlockUsingConnection(ctx, bs.Connection)
+	return bs.getOldestHistoryBlockUsingConnection(ctx, bs.Connection)
+}
+
+func (bs *Blocks) getOldestHistoryBlockUsingConnection(ctx context.Context, connection Connection) (entities.Block, error) {
+	block := &entities.Block{}
+	if err := pgxscan.Get(ctx, connection, block, `SELECT vega_time, height, hash
+		FROM blocks order by height asc limit 1`); err != nil {
+		return entities.Block{}, bs.wrapE(err)
+	}
+
+	return *block, nil
 }
 
 func GetOldestHistoryBlockUsingConnection(ctx context.Context, connection Connection) (entities.Block, error) {
```

Comment:
Do we need to keep the old `GetOldestHistoryBlockUsingConnection` and `GetLastBlockUsingConnection` below? That code duplication could be confusing and cause problems if we need to change anything in the future and forget to do both.

Reply:
yea it might not be needed, just didn't handle error exactly the same now.